src/main/java/by/gsu/epamlab/PercentDiscountPurchase.java

```diff
@@ -0,0 +1,31 @@
+package by.gsu.epamlab;
+
+
+public class PercentDiscountPurchase extends AbstractPurchase {
+    private static final int NUMBER_PURCHASES = 10;
+    private double percent;
+
+    public PercentDiscountPurchase(Product product, int numberUnits, double percent) {
+        super(product, numberUnits);
+        this.percent = percent;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("%s;%s", fieldsToString(), getCost());
```

Comment:
```suggestion
    protected String fieldsToString() {
        return String.format("%s;%.2f", super.fieldsToString(), percent);
```
аналогично в других подклассах.